torch/csrc/jit/codegen/cuda/kernel_expr_evaluator.cpp

```diff
@@ -41,8 +41,12 @@ void ExpressionEvaluator::bind(
 
 c10::optional<Int::ScalarType> ExpressionEvaluator::evaluate(const Val* value) {
   if (precomputed_integers_ && precomputed_integers_->ready()) {
-    return precomputed_integers_->getMaybeValueFor(value);
-  } else if (value->isScalar() && value->isConst()) {
+    if (precomputed_integers_->getMaybeValueFor(value).has_value()) {
```

Comment:
This case seems to go beyond expression evaluator. 

We are doing a DCE after allocating global buffers. [https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/lower2device.cpp#L512-L523](https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/lower2device.cpp#L512-L523)

We might also need to cleanup the global allocations if the grid reduction gets removed by DCE. 

Would you mind if we merge this PR as is to unblock and then I could do a follow up?

Reply:
Yeah, sounds good.